bin/benchmark_nc.py

```diff
@@ -0,0 +1,601 @@
+        fb = keep_only_digits(fb[0])
+
+        if fa < fb:
+            return -1
+        if fa == fb:
+            return 0
+        if fa > fb:
+            return 1
+
+    base = map(lambda x: os.path.abspath(x), maybe_inspect_zip(models))
+    base.sort(cmp=nsort)
+
+    return base
+
+def copy_tree(dir=None):
+    assert_valid_dir(dir)
+
+    sftp = ssh_conn.open_sftp()
+    try:
+        if stat.S_ISDIR(sftp.stat(dir).st_mode):
+            print('Directory already existent: %s' % dir)
+    except IOError:
+        print('Creating remote directory: %s' % dir)
+        sftp.mkdir(dir)
+
+    print('Copy files to remote')
+    for fname in os.listdir(dir):
+        fullpath = os.path.join(dir, fname)
+        local_stat  = os.stat(fullpath)
+        try:
+            remote_mode = sftp.stat(fullpath).st_mode
+        except IOError:
+            remote_mode = 0
+        if not stat.S_ISREG(remote_mode):
+            print('Copying %s ...' % fullpath)
+            remote_mode = sftp.put(fullpath, fullpath, confirm=True).st_mode
+        #else:
+        #    print('File exists: %s' % fullpath)
+
+        if local_stat.st_mode != remote_mode:
+            print('Setting proper remote mode: %s' % local_stat.st_mode)
+            sftp.chmod(fullpath, local_stat.st_mode)
+
+    sftp.close()
+
+def delete_tree(dir=None):
+    assert_valid_dir(dir)
+
+    sftp = ssh_conn.open_sftp()
+    if not stat.S_ISDIR(sftp.stat(dir).st_mode):
+        print('No remote directory: %s' % dir)
+    else:
+        print('Removing remote files')
+        for fname in sftp.listdir(dir):
+            fullpath = os.path.join(dir, fname)
+            remote_stat = sftp.stat(fullpath)
+            if stat.S_ISREG(remote_stat.st_mode):
+                print('Removing %s ...' % fullpath)
+                sftp.remove(fullpath)
+
+        print('Removing directory %s ...' % dir)
+        sftp.rmdir(dir)
+
+    sftp.close()
+
+def setup_tempdir(dir=None, models=[], wav=None, alphabet=None, binaries=None):
+    r'''
+    Copy models, libs and binary to a directory (new one if dir is None)
+    '''
+    if dir is None:
+        dir = tempfile.mkdtemp(suffix='dsbench')
+
+    sorted_models = all_files(models=models)
+    if binaries is None:
+        maybe_download_binaries(dir)
+    else:
+        print('Using local binaries: %s' % (binaries))
+        shutil.copy2(binaries, dir)
+    extract_tarball(dir)
+
+    filenames = map(lambda x: os.path.join(dir, os.path.basename(x)), sorted_models)
+    missing_models = filter(lambda x: not os.path.isfile(x), filenames)
+    if len(missing_models) > 0:
+        # If we have a ZIP file, directly extract it to the proper path
+        if is_zip_file(models):
+            print('Extracting %s to %s' % (models[0], dir))
+            zipfile.ZipFile(models[0]).extractall(path=dir)
+            print('Extracted %s.' % models[0])
+        else:
+            # If one model is missing, let's copy everything again. Be safe.
```

Comment:
That's for setting up the temporary dir

Reply:
I don't understand your reply. Where else do we copy those files?

Reply:
Okay, I misunderstood your point before. So that is to ensure there is no missing models, in case of interrupted run that may happen. It happened to me, I don't remember under which conditions precisely, but I had to add that code.

Reply:
Ah, makes sense.